### PR TITLE
[17.0][MIG] account_statement_import_file: migration to 17.0

### PR DESCRIPTION
**Pull request by manuelregidor**
Supersedes https://github.com/OCA/bank-statement-import/pull/649
MIgration + Tests added

### CODE REVIEW

**Comment by manuelregidor** [comment]
@HaraldPanten @ValentinVinagre

**Comment by HaraldPanten** [comment]
@hbrunn We have added some tests regarding your comment --> https://github.com/OCA/bank-statement-import/pull/649#pullrequestreview-2014602944

@pedrobaeza

File: account_statement_import_file/wizard/account_statement_import.py [hunk 1]
```diff
@@ -0,0 +1,358 @@
...
+        self.import_single_file(file_data, result)
+        logger.debug("result=%s", result)
+        if not result["statement_ids"]:
+            raise UserError(
+                _(
+                    "You have already imported this file, or this file "
+                    "only contains already imported transactions."
+                )
+            )
+        self.env["ir.attachment"].create(self._prepare_create_attachment(result))
```
Comment (pedrobaeza) [inline account_statement_import_file/wizard/account_statement_import.py]: @hbrunn's comment about using `attachment_ids` is still missing it seems.

Comment (manuelregidor) [reply account_statement_import_file/wizard/account_statement_import.py]: @pedrobaeza I will add it.

**Review by HaraldPanten — APPROVED** [review]
LGTM

Last comments have been applied. THX.

@pedrobaeza

File: account_statement_import_file/wizard/account_statement_import.py [hunk 2]
```diff
@@ -39,7 +39,13 @@ def _import_file(self):
                     "only contains already imported transactions."
                 )
             )
-        self.env["ir.attachment"].create(self._prepare_create_attachment(result))
+        attachment = self.env["ir.attachment"].create(
+            self._prepare_create_attachment(result)
+        )
+        statement = self.env["account.bank.statement"].browse(
+            result["statement_ids"][0]
```
Comment (hbrunn) [inline account_statement_import_file/wizard/account_statement_import.py]:
```suggestion
            result["statement_ids"]
```

Comment (ValentinVinagre) [reply account_statement_import_file/wizard/account_statement_import.py]: @manuelregidor

**Review by ValentinVinagre — APPROVED** [review]
LGTM 👍🏻

File: account_statement_import_file/wizard/account_statement_import.py (hunk 2 above)
Comment (manuelregidor) [reply account_statement_import_file/wizard/account_statement_import.py]: done

**Review by hbrunn — APPROVED** [review]
thanks!

**Pull request merged**